test/data-basic.jl

```diff
@@ -35,6 +35,35 @@
         @test isapprox(result["objective"], 16551.7; atol=1e0)
     end
 
+    @testset "In-place make_basic_network!" begin
+        data = PowerModels.parse_file("../test/data/matpower/case7_tplgy.m")
+        data_ = deepcopy(data)
+        
+        # Test #1: `make_basic_network` is non-destructive
+        data_basic = PowerModels.make_basic_network(data)
+        @test data == data_        # input data not modified
+        @test data_basic !== data  # we created a new dict
+
+        # Test #2: `make_basic_network!` is destructive and works in-place
+        data_basic = PowerModels.make_basic_network!(data)
+        @test data_copy === data  # in-place modification
```

Comment:
Here I only check that the returned dictionary is exactly the one that was passed as input.
Note the use of triple equal `===`